### DISCUSSION TITLE
Storybook and auth

### DISCUSSION THREAD

**Discussion by donnellyrankin** [vercel/community#2406] [Help] [answered]
### Summary

I would like to be able to create both ephemeral (for each branch)  and a permanent vercel environment to deploy storybook, behind auth. Is vercel authentication purely for next builds, or can it be utilized for envs set up for `npm run storybook`? What level of account would be needed for this functionality?

**Comment by amyegan** [comment, marked as answer]
Hi @donnellyrankin! [Vercel Authentication](https://vercel.com/docs/concepts/deployments/deployment-protection#vercel-authentication) is available on all plans and frameworks to control access to your Preview Deployments. You could use that for Previews, and then set up one of the many [other auth options](https://vercel.com/guides/application-authentication-on-vercel) to protect your Production Deployments. Or, if you're on a Pro plan, you could use the [Private Production Deployments](https://vercel.com/docs/concepts/deployments/deployment-protection#private-production-deployments) add-on.

**Reply by donnellyrankin** [reply]
Thank you very much!
